app/routes/datasets/versions.py

```diff
@@ -121,6 +131,9 @@ async def add_new_version(
 
     input_data["creation_options"] = creation_options
 
+    _ = input_data.pop(
```

Comment:
So this just throws it away for legacy versions? Unclear what the purpose of doing this is

Reply:
This is to avoid sending version metadata which is no longer valid for assets to `create_default_asset` function in the next step which will try to process any incoming `metadata` in `input_data`.